### PR TITLE
Add partial as checkbox argument & tests & doc

### PR DESCRIPTION
**Pull request by JulienVannier66**
### What does this PR do?
Add partial as checkbox argument. Clean up & adjust the css class for size. Update tests and documentation

Related to: https://github.com/upfluence/backlog/issues/2327

### What are the observable changes?
Peek 2023-01-30 12-02

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/checkbox.ts [hunk 1]
```diff
@@ -43,6 +44,16 @@ export default class OSSCheckbox extends Component<OSSCheckboxArgs> {
     return classes.join(' ');
   }
 
+  get fakeCheckboxClasses(): string {
```
Comment (Miexil) [inline addon/components/o-s-s/checkbox.ts:47]: I think we can remove the "fake" wording at least from the getter

File: addon/components/o-s-s/checkbox.ts [hunk 2]
```diff
@@ -43,6 +44,16 @@ export default class OSSCheckbox extends Component<OSSCheckboxArgs> {
...
+    const classes = ['upf-checkbox__fake-checkbox'];
+
+    if (this.args.partial) {
+      classes.push('upf-checkbox__fake-checkbox--partial');
```
Comment (Miexil) [inline addon/components/o-s-s/checkbox.ts:51]: partial prevails over checked right ?
maybe add a test "If checked is truthy and partial is truthy, the checkbox is partially checked" ?

**Review by Miexil — APPROVED** [review]
small comments, lgtm

File: tests/dummy/app/templates/application.hbs [hunk 3]
```diff
@@ -1,3 +1,16 @@
+<div class="fx-row fx-xalign-start fx-gap-px-10 margin-md">
+  OSS:Checkbox:
```
Comment (phndiaye) [inline tests/dummy/app/templates/application.hbs:2]: unfinished line

**Review by phndiaye — COMMENTED** [review]
small comment but lgtm

**Review by phndiaye — APPROVED** [review]

**Pull request merged**